connect/runtime/src/main/java/org/apache/kafka/connect/storage/FileOffsetBackingStore.java

```diff
@@ -73,36 +73,40 @@ public synchronized void stop() {
 
     @SuppressWarnings("unchecked")
     private void load() {
+        Object obj = null;
+
         try (SafeObjectInputStream is = new SafeObjectInputStream(Files.newInputStream(file.toPath()))) {
-            Object obj = is.readObject();
-            if (!(obj instanceof HashMap))
-                throw new ConnectException("Expected HashMap but found " + obj.getClass());
-            Map<byte[], byte[]> raw = (Map<byte[], byte[]>) obj;
-            data = new HashMap<>();
-            for (Map.Entry<byte[], byte[]> mapEntry : raw.entrySet()) {
-                ByteBuffer key = (mapEntry.getKey() != null) ? ByteBuffer.wrap(mapEntry.getKey()) : null;
-                ByteBuffer value = (mapEntry.getValue() != null) ? ByteBuffer.wrap(mapEntry.getValue()) : null;
-                data.put(key, value);
-                OffsetUtils.processPartitionKey(mapEntry.getKey(), mapEntry.getValue(), keyConverter, connectorPartitions);
-            }
+            obj = is.readObject();
         } catch (NoSuchFileException | EOFException e) {
```

Comment:
Could there be a case that earlier `NoSuchFileException` and `EOFEception` were no ops, but now `load` could throw  `ConnectException` because obj would be null and it would proceed the rest of the logic?